src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractWorkspacePullDiagnosticsHandler.cs

```diff
@@ -56,32 +51,17 @@ public void Dispose()
         _workspaceRegistrationService.LspSolutionChanged -= OnLspSolutionChanged;
     }
 
-    protected override async ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagnosticSourcesAsync(TDiagnosticsParams diagnosticsParams, RequestContext context, CancellationToken cancellationToken)
+    protected override ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagnosticSourcesAsync(TDiagnosticsParams diagnosticsParams, string? requestDiagnosticCategory, RequestContext context, CancellationToken cancellationToken)
     {
-        // If we're being called from razor, we do not support WorkspaceDiagnostics at all.  For razor, workspace
-        // diagnostics will be handled by razor itself, which will operate by calling into Roslyn and asking for
-        // document-diagnostics instead.
         if (context.ServerKind == WellKnownLspServerKinds.RazorLspServer)
-            return [];
-
-        Contract.ThrowIfNull(context.Solution);
-
-        var category = GetDiagnosticCategory(diagnosticsParams);
-
-        // TODO: Implement as extensibility point. https://github.com/dotnet/roslyn/issues/72896
-
-        if (category == PullDiagnosticCategories.Task)
-            return GetTaskListDiagnosticSources(context, GlobalOptions);
-
-        if (category == PullDiagnosticCategories.EditAndContinue)
-            return await EditAndContinueDiagnosticSource.CreateWorkspaceDiagnosticSourcesAsync(context.Solution, document => context.IsTracking(document.GetURI()), cancellationToken).ConfigureAwait(false);
-
-        // if this request doesn't have a category at all (legacy behavior, assume they're asking about everything).
-        if (category == null || category == PullDiagnosticCategories.WorkspaceDocumentsAndProject)
-            return await GetDiagnosticSourcesAsync(context, GlobalOptions, cancellationToken).ConfigureAwait(false);
+        {
+            // If we're being called from razor, we do not support WorkspaceDiagnostics at all.  For razor, workspace
+            // diagnostics will be handled by razor itself, which will operate by calling into Roslyn and asking for
+            // document-diagnostics instead.
+            return new([]);
+        }
 
-        // if it's a category we don't recognize, return nothing.
-        return [];
+        return _diagnosticSourceManager.CreateDiagnosticSourcesAsync(context, requestDiagnosticCategory, false, cancellationToken);
```

Comment:
named arg for 'false'.  it's meaning is not apparent in context.

Reply:
There are 23 `internal abstract class` in this project, i.e. seems to be a common pattern. I can refactor per your recommendation if required.